editor/edit-post/header/editor-actions/index.js

```diff
@@ -2,19 +2,16 @@
  * WordPress dependencies
  */
 import { MenuItemsGroup } from '@wordpress/components';
+import { applyFilters } from '@wordpress/hooks';
 import { __ } from '@wordpress/i18n';
 
-/**
- * Internal dependencies
- */
-import CopyContentButton from '../copy-content-button';
-
 export default function EditorActions() {
+	const children = applyFilters( 'editor.EditorActions.children', [] );
```

Comment:
It will be executed on every re-render, so it might be not the best strategy. It needs to be confirmed that it is fine as it is. There is also [withFilters HOC](https://github.com/WordPress/gutenberg/tree/master/components/higher-order/with-filters) which might be helpful. I will think about it a bit more tomorrow.

Reply:
Yeah, I didn't want to overthink the exact hooking mechanism used so we could discuss this. I just pushed f12cfd69, which simply stores the element in the module. It works `¯\_(ツ)_/¯`.

Reply:
This is cool, but I think we should call it more expressively with `editor.EditorActions.tools` and `editor.EditorActions.plugins` as the interface instead of `children`.

Reply:
That works, at least for `tools`. Does `plugins` necessarily belong under `EditorActions`?

Reply:
It's there in @jasmussen mockups.

Reply:
Sure, I didn’t mean visually. My question was really whether it should be handled by EditorActions or whether there should be both: EditorPlugins and EditorTools (the latter renamed from EditorActions). In any case, I have a local patch to make EA handle the two.

Reply:
Well, it mostly depends on what API we want folks to use. Would "tools" be open to anyone? Maybe "plugins" is the default if you don't specify, etc.

Reply:
Sharing how the dropdown looks like as it should drive the discussion in the right direction:
![screen shot 2018-01-19 at 09 31 00](https://user-images.githubusercontent.com/699132/35141560-93475034-fcfb-11e7-80dd-0fd767a234b0.png)

So we have 3 sections at the moment: `Editor`, `Settings` and `Tools`. It seems like we might want to add more sections if necessary. Question is if we really want to create `Plugins` section which could be also named `Others` or something along those lines. My point is that it is much nicer to have items grouped by their functionality rather than put everything created by the plugin developers in one group.

Reply:
This feels like an area that can be relatively easily optimized at a later time. While it's already getting a bit crowded, we have a number of features that we know we want to add to this menu. Perhaps we should be okay with adding these, and then agree to revisit this later on for optimization.

For example if we add many more settings and exporters, it might be good to keep all those sections. Or we might not add any more, and combine both the toolbar setting and the copy all into a single "Tools & Export" section. Or we might even look at moving the "Copy All Content" into a "Plugins" section.

Reply:
@karmatosed @jasmussen any thoughts?

Reply:
Same with hooks, you would just need to update the hook postfix to change the group, or we could consolidate `tools` & `export` into one group to follow your example.